### PR TITLE
[WP-448] Improve batchUid handling on upload

### PR DESCRIPTION
**Pull request by paveldhq**
- added logging with traces when faced submissions without batchUid on upload
 - added more strict logic to handle such issue on submission creation
 - added logic that tries to get batchUid if needed.

### CODE REVIEW

File: inc/Smartling/Helpers/TranslationHelper.php [hunk 1]
```diff
@@ -87,6 +89,20 @@ public function setMutilangProxy($mutilangProxy)
      */
     public function prepareSubmissionEntity($contentType, $sourceBlog, $sourceEntity, $targetBlog, $targetEntity = null)
     {
+        if (0 === (int)$targetBlog) {
```
Comment (dimitrystd) [inline inc/Smartling/Helpers/TranslationHelper.php]: Should we here check `$targetBlog` in `blogs` table and cover all possible errors in such way? what do you think?

Comment (paveldhq) [reply inc/Smartling/Helpers/TranslationHelper.php]: sounds cool, but not in table. will create a validator.

File: inc/Smartling/ApiWrapper.php [hunk 2]
```diff
@@ -697,15 +698,36 @@ public function retrieveBatch($profile, $jobId, $authorize = true, $updateJob =
...
-            $this->getLogger()->error(vsprintf('Can\'t create batch for a job "%s". Error: %s', [$jobId,
-                                                                                                 $e->formatErrors()]));
+            $this
+                ->getLogger()
+                ->error(
+                    vsprintf(
+                        'Can\'t create batch for a job "%s".\nProfile: %s\nError:%s.\nTrace:%s',
+                        [
+                            $jobId,
+                            base64_encode(serialize($profile->toArray(false))),
```
[the remark this answers is not in the document]
Comment (paveldhq) [reply inc/Smartling/ApiWrapper.php]: i wanted to have a full real dump for that case to try to reproduce if fires, that is why i keep all data.

**Comment by dimitrystd** [comment]
Reviewed splunk logs for this bug (empty batchUid). We have 2 log entries for the same error.
```
[2019-02-28 16:31:17] [ERROR] level="400", remote_log_datetime="2019-02-28 16:31:06", ctx_host="tinder-swipe-life-preprodgo-vipco_wpcli_0001", ctx_http_host="tinder-swipe-life-preprod.go-vip.co", ctx_moduleVersion="1.9.3", ctx_pluginVersion="1.9.3", ctx_requestId="5c780ca2107b1", ctx_projectId="49e58c0b6", ctx_loggerChannel="Smartling\ApiWrapper" Unexpected error occurred

[2019-02-28 16:31:17] [ERROR] level="400", remote_log_datetime="2019-02-28 16:31:06", ctx_host="tinder-swipe-life-preprodgo-vipco_wpcli_0001", ctx_http_host="tinder-swipe-life-preprod.go-vip.co", ctx_moduleVersion="1.9.3", ctx_pluginVersion="1.9.3", ctx_requestId="5c780ca2107b1", ctx_projectId="49e58c0b6", ctx_loggerChannel="Smartling\Base\SmartlingCore" Array
(
    [0] => Array
        (
            [key] => 
            [message] => Unexpected error occurred
            [details] => 
        )
)
```

**Pull request merged**